packages/client/src/subscribe/Subscriber.ts

```diff
@@ -185,6 +187,30 @@ export default class Subscriber implements Context {
         ]))
     }
 
+    async setSubscribeProxy(streamDefinition: StreamDefinition, nodeId: string): Promise<void> {
+        const streamPartId = await this.streamIdBuilder.toStreamPartID(streamDefinition)
+        await this.node.openSubscribeProxyConnectionOnStreamPart(streamPartId, nodeId)
+    }
+
+    async removeSubscribeProxy(streamDefinition: StreamDefinition, nodeId: string): Promise<void> {
+        const streamPartId = await this.streamIdBuilder.toStreamPartID(streamDefinition)
+        await this.node.closeSubscribeProxyConnectionOnStreamPart(streamPartId, nodeId)
+    }
+
+    async setSubscribeProxies(streamDefinition: StreamDefinition, nodeIds: string[]): Promise<void> {
+        const streamPartId = await this.streamIdBuilder.toStreamPartID(streamDefinition)
+        await Promise.allSettled([
+            ...nodeIds.map((nodeId) => this.node.openSubscribeProxyConnectionOnStreamPart(streamPartId, nodeId))
+        ])
+    }
+
+    async removeSubscribeProxies(streamDefinition: StreamDefinition, nodeIds: string[]): Promise<void> {
+        const streamPartId = await this.streamIdBuilder.toStreamPartID(streamDefinition)
+        await Promise.allSettled([
+            ...nodeIds.map(async (nodeId) => this.node.closeSubscribeProxyConnectionOnStreamPart(streamPartId, nodeId))
+        ])
+    }
+
```

Comment:
You know I feel like this stuff should probably go into its own file e.g. `subscribe/SubscriberProxy.ts`

e.g.
```ts
class SubscriberProxy {
    constructor(
        context: Context,
        private node: BrubeckNode,
        @inject(StreamIDBuilder) private streamIdBuilder: StreamIDBuilder,
    ) {
    
    }
    
    async setSubscribeProxy(streamDefinition: StreamDefinition, nodeId: string): Promise<void> {
        const streamPartId = await this.streamIdBuilder.toStreamPartID(streamDefinition)
        await this.node.openSubscribeProxyConnectionOnStreamPart(streamPartId, nodeId)
    }
    // etc
}
```
Your call though.